concent_api/core/validation.py

```diff
@@ -19,43 +20,70 @@ def validate_int_value(value):
         try:
             value = int(value)
         except (ValueError, TypeError):
-            raise Http400("Wrong type, expected a value that can be converted to an integer.")
+            raise Http400(
+                "Wrong type, expected a value that can be converted to an integer.",
+                error_code=ErrorCode.MESSAGE_VALUE_NOT_INTEGER,
+            )
     if value < 0:
-        raise Http400("Wrong type, expected non-negative integer but negative integer provided.")
+        raise Http400(
+            "Wrong type, expected non-negative integer but negative integer provided.",
+            error_code=ErrorCode.MESSAGE_VALUE_NOT_INTEGER,
+        )
     return value
 
 
 def validate_id_value(value, field_name):
     if not isinstance(value, str):
-        raise Http400("{} must be string.".format(field_name))
+        raise Http400(
+            "{} must be string.".format(field_name),
+            error_code=ErrorCode.MESSAGE_VALUE_NOT_STRING,
+        )
 
     if value == '':
-        raise Http400("{} cannot be blank.".format(field_name))
+        raise Http400(
+            "{} cannot be blank.".format(field_name),
+            error_code=ErrorCode.MESSAGE_VALUE_BLANK,
+        )
 
     if len(value) > MESSAGE_TASK_ID_MAX_LENGTH:
-        raise Http400("{} cannot be longer than {} chars.".format(field_name, MESSAGE_TASK_ID_MAX_LENGTH))
+        raise Http400(
+            "{} cannot be longer than {} chars.".format(field_name, MESSAGE_TASK_ID_MAX_LENGTH),
+            error_code=ErrorCode.MESSAGE_VALUE_WRONG_LENGTH,
```

Comment:
There is a problem, because this function validates currently both fields `task_id` and `subtask_id` but it can do any other field that requires this set of validators. Same for `validate_public_key` - we use it for `provider_public_key`, `requestor_public_key`, `client_public_key` and I guess it few others too.

I don't want to make error codes formatted, because I think that is what error messages are for. I also don't want to add extra if-ology only to determine error code in a function that can accept any field value as a parameter.

It would be possible to add field name ie. to `validate_subtask_price_task_to_compute` but then we will loose some kind of consistency, so I would like to avoid that.

Reply:
Oh I see that this function is just for validating the ID and nothing else. When I made this comment must have I assumed that it was something that validates a whole message.

Well, in that case there's probably not much that we can do here.

Though I'm more and more convinced that we should add `field` to the response in the future so that we can indicate which specific field has a problem. Then we could have more generic errors (like @dybi wanted) but still let the client watch for very specific errors.